src/utils/assets/assets.js

```diff
@@ -190,6 +190,54 @@ export function search({ AMId, query }, callback) {
 }
 
 /**
+ * Search for Assets with specified fields
+ * @function fieldsSearch
+ * @memberof module:api.Assets
+ * @static
+ * @param {object} params - object of parameters:
+ * @param {number} [params.AMId] - Asset Manager ID of the Assets to search over. If omitted, you must send assetManagerIds in the search query with at least one value
+ * @param {number} [params.assetId] -Asset ID of the asset
+ * @param {string} [params.field] -Fields should be returned 
+ * Available fields are:
+ * <li>assetManagerIds (Required if AMId param is omitted)</li>
+ * <li>clientIds</li>
+ * <li>assetStatuses</li>
+ * <li>assetIds</li>
+ * <li>referenceTypes</li>
+ * <li>referenceValues</li>
+ * <li>assetIssuerIds</li>
+ * <li>assetClasses</li>
+ * <li>assetTypes</li>
+ * e.g. `{ AMIds: [1], assetIds: [1], fields: [assetType, description] }`
+ * @param {function} callback - Called with two arguments (error, result) on completion. `result` is an array of Assets or a single Asset instance
+ * @returns {Promise|null} If no callback supplied, returns a Promise that resolves with an array of Assets or a single Asset instance
+ */
+export function fieldsSearch({ AMIds, assetIds, fields }, callback) {
+  const params = {
+    AMaaSClass: 'assets',
+    AMIds,
```

Comment:
You cannot have an array of AMIds.

This is what the `searchData` function does:

It accepts an object of the following shape:

```javascript
const params = {
  AMaaSClass: 'assets', // string
  AMId: 1, // number
  query: [{
    assetIds: ['asset1', 'asset2'],
    assetClasses: ['Equity', 'Future'],
    // other fields
  }]
}
```
It will convert these parameters into a HTTP call like this:

```javascript
'api.amaas.com/v1.0/asset/assets/1?assetIds=asset1,asset2&assetClasses=Equity,Future&camelcase=true'

// .../asset/assets/... This comes from AMaaSClass: 'assets';

// .../1/... This comes from AMId: 1 in params;

// ...assetIds=asset1,asset2... This comes from the first member of the query array { assetIds: ['asset1', 'asset2'] };

// ...assetClasses=Equity,Future... This comes from the second member of the query array { assetClasses: ['Equity', 'Future'] };
```
For this task, `fields` becomes another search parameter, meaning the `params.query` array should look something like:

```javascript
params.query = [
  // other search parameters
  { fields: ['references', 'comments'] }
]
```

This means the returned Asset will not contain the `links` object, since we did not specify it in `fields`.

Currently for assets this will work, but for transactions and positions you will need to modify the endpoint itself because it does not currently know what to do with the `fields` query parameter.